src/Core/src/Credentials/IniFileProvider.php

```diff
@@ -92,6 +93,15 @@ private function getCredentialsFromProfile(array $profilesData, string $profile,
             return $this->getCredentialsFromRole($profilesData, $profileData, $profile, $circularCollector);
         }
 
+        $existsSsoClient = class_exists(SsoClient::class);
+        $hasProfileSsoLegacyConfig = isset($profileData[IniFileLoader::KEY_SSO_START_URL]);
+
+        if (!$existsSsoClient && $hasProfileSsoLegacyConfig) {
```

Comment:
I suggest doing the class_exists check only when needed:

```php
if (isset($profileData[IniFileLoader::KEY_SSO_START_URL])) {
    if (class_exists(SsoClient::class)) {
        return $this->getCredentialsFromLegacySso($profileData, $profile);
    }

    $this->logger->warning('The profile "{profile}" contains SSO (legacy) config but the "async-aws/sso" package is not installed.', ['profile' => $profile]);

    return null;
}
```

I'm also returning `null` to avoid reaching the info message saying no credentials are found. And I updated slightly the warning message.

Reply:
Looks better. Thx, I adjusted it